NuclearData/materialMenu_mod.f90

```diff
@@ -285,10 +288,13 @@ subroutine init_materialItem(self, name, idx, dict)
     ! Return to initial state
     call self % kill()
 
-    ! Load easy components c
+    ! Load easy components properties
     self % name = name
     self % matIdx = idx
-    call dict % get(self % T,'temp')
+
+    ! If material temperature is present TMS is switched on
+    if (dict % isPresent('temp')) self % hasTMS = .true.
```

Comment:
If all of the above comment can't be addressed, even just having a separate flag for TMS and temperature would be an improvement for future developments I think.